src/Compilers/CSharp/Test/Semantic/Semantics/StaticNullChecking_Members.cs

```diff
@@ -0,0 +1,1006 @@
+                //         o = c.A.A; // 5
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "c.A").WithLocation(32, 13),
+                // (32,13): warning CS8601: Possible null reference assignment.
+                //         o = c.A.A; // 5
+                Diagnostic(ErrorCode.WRN_NullReferenceAssignment, "c.A.A").WithLocation(32, 13),
+                // (33,13): warning CS8601: Possible null reference assignment.
+                //         o = c.B.P; // 5
+                Diagnostic(ErrorCode.WRN_NullReferenceAssignment, "c.B.P").WithLocation(33, 13));
+        }
+
+        [Fact]
+        public void ModifyMembers_Struct()
+        {
+            var source =
+@"#pragma warning disable 0649
+struct A
+{
+    internal object? F;
+}
+struct B
+{
+    internal A A;
+    internal object? G;
+}
+class Program
+{
+    static void F()
+    {
+        object o;
+        B b = new B();
+        b.G = new object();
+        b.A.F = new object();
+        o = b.G; // 1
+        o = b.A.F; // 1
+        b.A = default(A);
+        o = b.G; // 2
+        o = b.A.F; // 2
+        b = default(B);
+        o = b.G; // 3
+        o = b.A.F; // 3
+    }
+}";
+            var comp = CreateStandardCompilation(source, parseOptions: TestOptions.Regular8);
+            comp.VerifyDiagnostics(
+                // (23,13): warning CS8601: Possible null reference assignment.
+                //         o = b.A.F; // 2
+                Diagnostic(ErrorCode.WRN_NullReferenceAssignment, "b.A.F").WithLocation(23, 13),
+                // (25,13): warning CS8601: Possible null reference assignment.
+                //         o = b.G; // 3
+                Diagnostic(ErrorCode.WRN_NullReferenceAssignment, "b.G").WithLocation(25, 13),
+                // (26,13): warning CS8601: Possible null reference assignment.
+                //         o = b.A.F; // 3
+                Diagnostic(ErrorCode.WRN_NullReferenceAssignment, "b.A.F").WithLocation(26, 13));
+        }
+
+        // PROTOTYPE(NullableReferenceTypes): Handle struct properties that are not auto-properties.
+        [Fact(Skip = "Struct property not auto-property")]
+        public void ModifyMembers_StructPropertyExplicitAccessors()
+        {
+            var source =
+@"#pragma warning disable 0649
+struct S
+{
+    private object? _p;
+    internal object? P { get { return _p; } set { _p = value; } }
+}
+class C
+{
+    S F;
+    void M()
+    {
+        object o;
+        o = F.P; // 1
+        F.P = new object();
+        o = F.P; // 2
+    }
+}";
+            var comp = CreateStandardCompilation(source, parseOptions: TestOptions.Regular8);
+            comp.VerifyDiagnostics(
+                // (13,13): warning CS8601: Possible null reference assignment.
+                //         o = F.P; // 1
+                Diagnostic(ErrorCode.WRN_NullReferenceAssignment, "F.P").WithLocation(13, 13));
+        }
+
+        [Fact]
+        public void ModifyMembers_StructPropertyFromMetadata()
```

Comment:
>ModifyMembers_StructPropertyFromMetadata [](start = 20, length = 40)

Do we have similar test for scenario when the function is defined in source?

Reply:
Added.